lib/beacon-verification.ts

```diff
@@ -4,15 +4,23 @@ import {ChainedBeacon, ChainInfo, isChainedBeacon, isUnchainedBeacon, Randomness
 async function verifyBeacon(chainInfo: ChainInfo, beacon: RandomnessBeacon): Promise<boolean> {
     const publicKey = chainInfo.public_key
 
+    let message: Uint8Array
     if (isChainedBeacon(beacon, chainInfo)) {
-        return bls.verify(beacon.signature, await chainedBeaconMessage(beacon), publicKey)
-    }
-    if (isUnchainedBeacon(beacon, chainInfo)) {
-        return bls.verify(beacon.signature, await unchainedBeaconMessage(beacon), publicKey)
+        message = await chainedBeaconMessage(beacon)
+
+    } else if (isUnchainedBeacon(beacon, chainInfo)) {
+        message = await unchainedBeaconMessage(beacon)
+    } else {
+        console.error(`Beacon type ${chainInfo.schemeID} was not supported`)
+        return false
     }
 
-    console.error(`Beacon type ${chainInfo.schemeID} was not supported or the fields of the provided beacon did not match the requirements`)
-    return false
+    const signatureVerifies = await bls.verify(beacon.signature, message, publicKey)
+    if (!signatureVerifies || !await randomnessIsValid(beacon)) {
```

Comment:
I'd argue it's easier to read the De Morgan version:
```suggestion
    if (!(signatureVerifies && await randomnessIsValid(beacon))) {
```